src/libraries/Microsoft.PowerFx.Interpreter/Marshal/ITypeMarshallerProvider.cs

```diff
@@ -47,4 +47,14 @@ public interface ITypeMarshaller
         /// </remarks>
         public FormulaValue Marshal(object value);
     }
+
+    /// <summary>
+    /// Try to marshal a dynamic type such as a Dictionary, JObject, or DataTable.
+    /// Unlike <see cref="ITypeMarshaller"/>, the static type does not infer the runtime FormulaType. 
```

Comment:
Why is this the case?

Reply:
In a strong typing scenario, different JOBjects can yield different RecordValues - we don't know until we inspect the JObject's properties. 

IDynamicTypeMarshaller is letting you get a strong type for things like JOBject (rather than having to marshal as UntypedObject)

Reply:
(I'll update the comment)